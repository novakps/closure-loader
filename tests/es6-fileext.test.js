const path = require('path');
const execBundle = require('./helpers/execBundle');

const config = {
    module: {
        rules: [
            {
                test: /legacy-code\/.*\.(js|es6)$/,
                use: [
                    {
                        loader: require.resolve('..'),
                        options: {
                            paths: [
                                path.join(
                                    __dirname,
                                    'fixtures',
                                    'es6-fileext',
                                    'legacy-code',
                                ),
                            ],
                            es6mode: true,
                            fileExt: '.{js,es6}',
                            watch: false,
                        },
                    },
                ],
            },
        ],
    },
};

test('detects modules in files matching fileExt', async () => {
    const result = await execBundle('es6-fileext', config);
    expect(result.code).toBe(0);
    expect(result.stdout).toMatchSnapshot();
});
