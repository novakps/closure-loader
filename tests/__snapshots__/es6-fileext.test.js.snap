// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detects modules in files matching fileExt 1`] = `
"This should contain the empty submodule: { value: 'This is a module', submodule: {} }
Call via default export:
this is some.legacy.namespace.myFn
some.legacy.namespace.otherModule.text = This is from otherModule.
Call via named export:
this is some.legacy.namespace.myFn
some.legacy.namespace.otherModule.text = This is from otherModule.
Some var: 42"
`;
